src/inet/TCPEndPoint.cpp

```diff
@@ -1648,8 +1666,9 @@ CHIP_ERROR TCPEndPoint::DoClose(CHIP_ERROR err, bool suppressCallback)
                     ChipLogError(Inet, "SO_LINGER: %d", errno);
             }
 
-            if (mSocket.Close() != 0 && err == CHIP_NO_ERROR)
-                err = chip::System::MapErrorPOSIX(errno);
+            CHIP_ERROR status = mSocket.Close();
+            if (status != CHIP_NO_ERROR && err == CHIP_NO_ERROR)
```

Comment:
Could simplify to:
```suggestion
            if (err == CHIP_NO_ERROR)
```
in that if `err == CHIP_NO_ERROR` and `status == CHIP_NO_ERROR` the assignment is still fine.